sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -473,8 +478,8 @@ class GpuHashAggregateIterator(
 
       override def next(): ColumnarBatch = {
         // batches coming out of the sort need to be merged
-        withResource(keyBatchingIter.next()) { batch =>
-          computeAggregate(batch, mergeSortedHelper)
+        closeOnExcept(keyBatchingIter.next()) { batch =>
+          computeAggregateAndClose(batch, mergeSortedHelper)
         }
```

Comment:
Double-close issue again.
```suggestion
        computeAggregateAndClose(keyBatchingIterator.next(), mergeSortedHelper)
```